safe/gui/tools/minimum_needs/test/test_minimum_needs_tool.py

```diff
@@ -52,24 +56,60 @@ def tearDown(self):
             if os.path.exists(path):
                 os.remove(path)
 
-    # This test is expected to fail until we fix the minimum needs calculator
-    # in InaSAFE V4.
-    @unittest.expectedFailure
     def test_minimum_needs(self):
         """Test behaviour of the minimum needs function.
         """
         dialog = NeedsCalculatorDialog(PARENT)
-        layer = shapefile_path
+        layer = QgsVectorLayer(
+            shapefile_path,
+            os.path.basename(shapefile_path),
+            'ogr')
+        if layer.isValid():
+            layer = dialog.prepare_new_layer(layer)
         attribute = 'displaced'
-        new_layer = dialog.minimum_needs(layer, attribute)
-        assert new_layer is not None
-        attributes = {
-            'Drinking Water': 17500,
-            'Family Kits': 200,
-            'Rice': 2800,
-            'Toilets': 50,
-            'Clean Water': 67000}
-        self.assertDictEqual(attributes, dict(new_layer.data[0]))
+        layer.keywords = {
+            'layer_purpose': layer_purpose_aggregation['key'],
+            'inasafe_fields': {displaced_field['key']: attribute}}
+        rename_remove_inasafe_fields(layer)
+
+        # run minimum needs function
+        dialog.minimum_needs(layer)
+        assert layer is not None
+        field_names = [field.name() for field in layer.pendingFields()]
+        for feature in layer.getFeatures():
+            value = [attribute for attribute in feature.attributes()]
+
+        actual_attributes = dict(zip(field_names, value))
+
+        expected_attributes = {
+            'displaced': 1000,
+            'minimum_needs__rice': 2800,
+            'minimum_needs__drinking_water': 17500,
+            'minimum_needs__clean_water': 67000,
+            'minimum_needs__family_kits': 200,
+            'minimum_needs__toilets': 50}
+
+        self.assertDictEqual(expected_attributes, actual_attributes)
+
+    def test_ok_button(self):
+        """Test behaviour of Ok button
+        """
+        # Test Ok button without any input in the combo box
+        dialog = NeedsCalculatorDialog(PARENT)
+        ok_button = dialog.button_box.button(QtGui.QDialogButtonBox.Ok)
+        self.assertFalse(ok_button.isEnabled())
+
+        input_layer = QgsVectorLayer(
+            shapefile_path,
+            os.path.basename(shapefile_path),
+            'ogr')
+        QgsMapLayerRegistry.instance().addMapLayers([input_layer])
```

Comment:
I did this because I need to set the specific layer to the Qgis combo box (line 110). setLayer function will not set the layer if there is no layer on the canvas, I guess. I don't know if you have any other brilliant idea.